presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -42,45 +44,61 @@ protected void assertInvalidFunction(String projection, Type expectedType, Strin
     {
         try {
             assertFunction(projection, expectedType, null);
-            fail("Expected to throw an INVALID_FUNCTION_ARGUMENT exception with message " + message);
         }
         catch (PrestoException e) {
             assertEquals(e.getErrorCode(), INVALID_FUNCTION_ARGUMENT.toErrorCode());
             assertEquals(e.getMessage(), message);
+            return;
         }
+        fail("Expected to throw an INVALID_FUNCTION_ARGUMENT exception with message " + message);
```

Comment:
I'm sorry. I wasn't thinking right. I somehow believed that the original code was wrong. And changing the order would make it correct. But as I look at it again. That really is not the case. I'm sorry about it. If you agree, feel free to revert it.